tests/docs/test_run_generate_dataset_example.py

```diff
@@ -0,0 +1,91 @@
+import textwrap
+import unittest
+
+import pytest
+
+from . import test_complete_examples
+
+
+def test_run_generate_dataset_example(tmp_path):
+    example = test_complete_examples.DatasetDefinitionExample(
+        path="test",
+        fence_number=1,
+        source=textwrap.dedent(
+            """\
+            from ehrql import create_dataset
+            from ehrql.tables.beta.tpp import patients
+
+            dataset = create_dataset()
+            dataset.define_population(patients.exists_for_patient())
+            """
+        ),
+    )
+    test_complete_examples.test_ehrql_generate_dataset_example(tmp_path, example)
+
+
+def test_run_generate_dataset_example_failing(tmp_path):
+    example = test_complete_examples.DatasetDefinitionExample(
+        path="test",
+        fence_number=1,
+        source=textwrap.dedent(
+            """\
+            from ehrql import create_dataset
+
+            dataset = create_dataset()
+            dataset.define_population(not_a_function())
+            """
+        ),
+    )
+    with pytest.raises(test_complete_examples.DatasetDefinitionTestError) as exc_info:
+        test_complete_examples.test_ehrql_generate_dataset_example(tmp_path, example)
+    assert type(exc_info.value) is test_complete_examples.DatasetDefinitionTestError
```

Comment:
comment: not currently in the Pytest docs at time of writing, but [mentioned in this PR](https://github.com/pytest-dev/pytest/pull/11578)